src/main/java/org/sasanlabs/service/vulnerability/commandInjection/CommandInjection.java

```diff
@@ -204,20 +209,22 @@ private StringBuilder getResponseFromPingCommand(String ipAddress, Supplier<Bool
     public ResponseEntity<GenericVulnerabilityResponseBean<String>> getVulnerablePayloadLevel6(
             @RequestParam(IP_ADDRESS) String ipAddress)
             throws ServiceApplicationException, IOException {
-
+        Supplier<Boolean> condition =
+                getSupplier(
+                        StringUtils.isNotBlank(ipAddress)
+                                && (IP_ADDRESS_PATTERN.matcher(ipAddress).matches()
+                                        || ipAddress.contentEquals("localhost")));
         return new ResponseEntity<GenericVulnerabilityResponseBean<String>>(
                 new GenericVulnerabilityResponseBean<String>(
-                        this.getResponseFromPingCommand(
-                                        ipAddress,
-                                        () ->
-                                                ipAddress != null
-                                                        && (IP_ADDRESS_PATTERN
-                                                                        .matcher(ipAddress)
-                                                                        .matches()
-                                                                || ipAddress.contentEquals(
-                                                                        "localhost")))
-                                .toString(),
-                        true),
+                        this.getResponseFromPingCommand(ipAddress, condition).toString(), true),
                 HttpStatus.OK);
     }
+
+    /* In order to test getResponseFromPingCommand in unit test
+     * and verify with mockito to check condition is true or false
+     * getSupplier is extracted to an independent method
+     */
+    Supplier<Boolean> getSupplier(boolean condition) {
```

Comment:
I think this is not required. While checking the condition in mockito you need not to compare the 2 suppliers instead you can check condition.get() to be true or false.